ir/src/depgraph.rs

```diff
@@ -0,0 +1,421 @@
+use filetime::FileTime;
+use serde::{Deserialize, Serialize};
+
+use std::{
+    collections::{HashMap, HashSet},
+    fs, io,
+    path::{Path, PathBuf},
+    vec,
+};
+
+/// Helps identify changes in a set of files.
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+#[serde(from = "DepGraphSerdeRepr", into = "DepGraphSerdeRepr")]
+pub struct DepGraph {
+    entries: HashMap<PathBuf, PathState>,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+enum PathState {
+    Seen { mtime: FileTime, size: u64 },
```

Comment:
noob question: if we already track the modified time of a file, why do we need also bother checking its size? Can the size change without file modified time changing as well?